File: python/phase 1-condtional_thinking/q11_valid_triangle.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 2: Nested If & Multiple Conditions
Question 11: Take three sides and check if they form a valid triangle.

Mentor Tips (Python 3.14+):
- Triangle inequality theorem states that sum of any two sides must be strictly greater than the third side.
"""

def solve(s1: float, s2: float, s3: float) -> bool:
    # TODO: Implement this method to solve the question
    if s1 + s2 > s3 and s1 + s3 > s2 and s2 + s3 > s1:
        return True
    else:
        return False

def main():
    s1 = float(input("Side 1: "))
    s2 = float(input("Side 2: "))
    s3 = float(input("Side 3: "))
    print(f"Valid triangle: {solve(s1, s2, s3)}")

if __name__ == "__main__":
    main()

File: java/src/phase 1-conditional_thinking_java/Q11ValidTriangle.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 2: Nested If & Multiple Conditions
 * Question 11: Take three sides and check if they form a valid triangle.
 * 
 * Mentor Tips (Java 26):
 * - Triangle inequality theorem states that sum of any two sides must be strictly greater than the third side.
 */
public class Q11ValidTriangle {

    public static boolean solve(double s1, double s2, double s3) {
        // TODO: Implement this method to solve the question
        return false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Side 1: "); double s1 = scanner.nextDouble();\n        System.out.print("Side 2: "); double s2 = scanner.nextDouble();\n        System.out.print("Side 3: "); double s3 = scanner.nextDouble();\n        System.out.println("Valid triangle: " + solve(s1, s2, s3));
        scanner.close();
    }
}
